iOS/UI/Swipeable Input/MessageDeliveryStatusUIViewRepresentable.swift

```diff
@@ -0,0 +1,114 @@
+        super.initializeSubviews()
+
+        let keypath = AnimationKeypath(keys: ["**", "Color"])
+        let colorProvider = ColorValueProvider(ThemeColor.D1.color.lottieColorValue)
+
+        self.addSubview(self.sendStatus)
+        self.sendStatus.currentProgress = 0
+        self.sendStatus.setValueProvider(colorProvider, keypath: keypath)
+        self.sendStatus.contentMode = .scaleAspectFit
+
+        self.addSubview(self.readStatusView)
+        self.readStatusView.currentProgress = 1
+        self.readStatusView.setValueProvider(colorProvider, keypath: keypath)
+        self.readStatusView.animationSpeed = 0.25
+        self.readStatusView.contentMode = .scaleAspectFit
+
+        let errorColorProvider = ColorValueProvider(ThemeColor.red.color.lottieColorValue)
+
+        self.addSubview(self.errorStatusView)
+        self.errorStatusView.currentProgress = 0
+        self.errorStatusView.setValueProvider(errorColorProvider, keypath: keypath)
+        self.errorStatusView.contentMode = .scaleAspectFit
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+
+        self.readStatusView.expandToSuperviewSize()
+        self.sendStatus.expandToSuperviewSize()
+        self.errorStatusView.expandToSuperviewSize()
+    }
+
+    private var previousMessage: Messageable?
+    private var previousStatus: DeliveryStatus?
+
+    func update(with message: Messageable, deliveryStatus: DeliveryStatus) {
+        // Ignore redundant states
+        guard message.id != self.previousMessage?.id
+                || deliveryStatus != self.previousStatus else { return }
+
+        defer {
+            self.previousMessage = message
+            self.previousStatus = deliveryStatus
+        }
+
+        self.readStatusView.isVisible = false
+        self.sendStatus.isVisible = false
+        self.errorStatusView.isVisible = false
+
+        switch deliveryStatus {
+        case .sending:
+            break
+        case .sent:
+            if !message.isFromCurrentUser {
+                self.readStatusView.isVisible = true
+                self.readStatusView.currentProgress = 1
+            } else {
+                self.sendStatus.isVisible = true
+                if self.previousStatus == .sending {
+                    self.sendStatus.stop()
+                    self.sendStatus.play()
+                } else {
+                    self.sendStatus.currentProgress = 1
+                }
+            }
+        case .reading:
+            self.readStatusView.isVisible = true
+            self.readStatusView.stop()
+            self.readStatusView.play(fromProgress: 1, toProgress: 0, loopMode: .playOnce)
+        case .read:
+            self.readStatusView.isVisible = true
+            self.readStatusView.currentProgress = 0
+        case .error:
+            self.errorStatusView.isVisible = true
+            self.errorStatusView.stop()
+            self.errorStatusView.play()
+        }
```

Comment:
Playing animations, or snapping to animation positions based upon the message state.